legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/dbTestRunner/shared.pure

```diff
@@ -54,19 +55,57 @@ function meta::relational::dbTestRunner::collectTests(pack: Package[1]): Functio
+{
+  if($config.connection->isEmpty(),
+    |  
+        println('**** Warning **** : No connection available for given dbtype, Skipping execution of setup sqls');
+        false;,
+    |   
+        let setupSqls = meta::alloy::service::execution::setUpDataSQLs($records, $db);
+
+        let con = $config.connection->cast(@RelationalDatabaseConnection)->toOne();
+        let connWithDb= ^$con(element= $db);
+
+        $setupSqls->executeInRelationalDb($connWithDb,meta::relational::extension::relationalExtensions());
+        println('setup done');
+        true;
+   )   
+}
+
 function meta::relational::dbTestRunner::executeInRelationalDb(sqls:String[*], conn:RelationalDatabaseConnection[1], extensions:meta::pure::extension::Extension[*]):Boolean[1]
 {
   meta::legend::test::mayExecuteLegendTest
      (
        {clientVersion, serverVersion, serializationKind, host, port |
-         let fStr = 'meta::protocols::pure::'+$clientVersion+'::invocation::execution::executeInRelationalDb::executeInRelationalDb_String_MANY__RelationalDatabaseConnection_1__String_1__Integer_1__Extension_MANY__Boolean_1_';
-         let xf = $fStr->pathToElement()->cast(@Function<{String[*], RelationalDatabaseConnection[1], String[1], Integer[1], meta::pure::extension::Extension[*]->Boolean[1]}>);
-         let res= $xf->evaluate(list($sqls)->concatenate([$conn, $host, $port]->map(v|list($v)))->concatenate(list($extensions)));
-         $res->cast(@Boolean)->toOne();
+            let fStr = 'meta::protocols::pure::'+$clientVersion+'::invocation::execution::executeInRelationalDb::executeInRelationalDb_String_MANY__RelationalDatabaseConnection_1__String_1__Integer_1__Extension_MANY__Boolean_1_';
+            let xf = $fStr->pathToElement()->cast(@Function<{String[*], RelationalDatabaseConnection[1], String[1], Integer[1], meta::pure::extension::Extension[*]->Boolean[1]}>);
+            let res= $xf->evaluate(list($sqls)->concatenate([$conn->toOne(), $host, $port]->map(v|list($v)))->concatenate(list($extensions)));
+            $res->cast(@Boolean)->toOne();
          },
        { |
          println('**** Warning **** : Legend Engine Server not available');
          false;
        }
      );
+}
+
+function meta::relational::dbTestRunner::executeViaPlan<T|m>(f:FunctionDefinition<{->T[m]}>[1], m:Mapping[1], db:meta::relational::metamodel::Database[1], config:DbTestConfig[1], extensions:meta::pure::extension::Extension[*]):meta::pure::mapping::Result<T|m>[0..1]
+{
+  if($config.connection->isEmpty(),
+     | 
+       println('**** Warning **** : No connection available for given dbtype, Skipping execution of plan');
+       let pureRuntime = ^Runtime(connections= ^TestDatabaseConnection(element = $db, type= $config.dbType));
+
+       let basicPlan = if($m.name->isEmpty(),
```

Comment:
what does this mean?

Reply:
it can be a case where we dont supply any mapping/runtime. ( going to be supported in engine as far as I understand)